src/components/ssi/AuthenticationModal/AuthenticationQR.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { Component } from 'react'
+import { BallTriangle } from 'react-loader-spinner'
+import {
+  AuthorizationResponseStateStatus,
+  AuthStatusResponse,
+  GenerateAuthRequestURIResponse
+} from './auth-model'
+import {
+  CreateElementArgs,
+  QRType,
+  URIData,
+  ValueResult
+} from '@sphereon/ssi-sdk.qr-code-generator'
+
+import { AuthorizationResponsePayload } from '@sphereon/did-auth-siop'
+import agent from '@components/ssi/AuthenticationModal/agent'
+
+export type AuthenticationQRProps = {
+  onAuthRequestRetrieved: () => void
+  onSignInComplete: (payload: AuthorizationResponsePayload) => void
+  setQrCodeData: (text: string) => void
+}
+
+export interface AuthenticationQRState {
+  authRequestURIResponse?: GenerateAuthRequestURIResponse
+  qrCode?: JSX.Element
+}
+
+export default class AuthenticationQR extends Component<AuthenticationQRProps> {
+  state: AuthenticationQRState = {}
+
+  private registerStateSent = false
+  private refreshTimerHandle?: NodeJS.Timeout
+  private qrExpirationMs = 0
+  private timedOutRequestMappings: Set<AuthenticationQRState> =
+    new Set<AuthenticationQRState>()
+
+  private _isMounted = false
+
+  private readonly definitionId =
```

Comment:
Please add some code to make sure that the definition cannot be undefined

Reply:
Interesting... The constant is not used, the definition id comes from the authentication response:
`{
  "correlationId": "b86b791d-83c8-4ec9-b84b-265b5194e958",
  "definitionId": "sphereon",
  "authRequestURI": "openid-vc://?request_uri=https%3A%2F%2Fssi-backend.sphereon.com%2Fsiop%2Fdefinitions%2Fsphereon%2Fauth-requests%2Fb86b791d-83c8-4ec9-b84b-265b5194e958",
  "authStatusURI": "https://ssi-backend.sphereon.com/webapp/auth-status"
}`

Reply:
No, the only reason that happens is because the REST client is configured with that env var, and thus that ensures you will see the definition in request/responses from that moment on.

As the REST client can also work in SIOP only flows (so no OID4VP), the definitionId is optional for the REST client (or it should be). That automatically means you will have to check it in environments where OID4VP has to be used.

Reply:
Ping @maikel-maas

Reply:
Fixed!